src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -14,21 +25,26 @@ export const TodoFilter = () => (
       <input
         data-cy="searchInput"
         type="text"
+        value={query}
         className="input"
         placeholder="Search..."
+        onChange={(event) => setQuery(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
+      { query !== '' && (
```

Comment:
```suggestion
      { query && (
```

<img width="114" alt="image" src="https://user-images.githubusercontent.com/99331793/195176821-2188222f-580d-4d4c-a7ca-f93f448d0f02.png">